src/scripts/main.js

```diff
@@ -1,3 +1,36 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+
+tree.addEventListener('click', (e) => {
+  if (e.target.tagName !== 'SPAN') {
+    return;
+  }
+
+  const subTree = e.target.nextElementSibling;
+
+  if (subTree && subTree.tagName === 'UL') {
+    subTree.hidden = !subTree.hidden;
+  }
+});
+
+const addSpansToHeaders = (p) => {
+  for (const li of p.querySelectorAll('li')) {
+    const headerText = li.firstChild;
```

Comment:
Ensure that `li.firstChild` is always a text node. If the structure of the HTML changes, this assumption might not hold, leading to unexpected behavior.